podaac/merger/merge_worker.py

```diff
@@ -160,15 +173,21 @@ def _run_worker(in_queue, out_queue, max_dims, var_info):
             origin_dataset.set_auto_maskandscale(False)
 
             for var_path, var_meta in var_info.items():
+
                 ds_group, var_name = resolve_group(origin_dataset, var_path)
                 ds_var = ds_group.variables[var_name]
-
                 resized_arr = resize_var(ds_var, var_meta, max_dims)
 
+                # Limit to how much memory we allocate to 60 MB
+                while memory_limit.value + resized_arr.nbytes > 60000000 and not out_queue.empty():
```

Comment:
Should this value (`60000000`) be an environment variable so we could change it at some point in the future? How does this affect the processing, does it just take longer? I'm wondering if we are at risk of not being able to use concise on certain data because we're limiting ourselves to 60MB? Like, could there be a dataset that won't fit into this 60MB and therefore wouldn't be able to be processed?

Reply:
1) I can put in to to get an env variable or get a default was hoping we can set it via harmony somewhere but doesn't seem like we can change the shared memory at the moment
2) The current running version of kubernetes doesn't have way to set the size, we can mount memory into shared memory but ends up taking half the memory of the kubernetes pod. There is a fix for it in later version of kubernetes. Another thing that harmony needs to add is `LocalStorageCapacityIsolation` which im not sure what that is.
3) It doesn't take any longer as concise is 2 process one read one write, and it will finish when the write is done and read is faster than write, so even if we throttle the read the whole thing will only finish as fast as the write process.
4) We are limited to the 60 mb as if there are variables that are bigger than that it will try to write to memory and crash.
5) Another option is to go with single core processing so we don't use shared memory until we have a solution to set the shared memory size limit in harmony.